1.5/Source/PawnStorages/PawnStorages/JobDriver_Release.cs

```diff
@@ -46,11 +46,19 @@ public override IEnumerable<Toil> MakeNewToils()
         {
             Pawn actor = release.actor;
             CompPawnStorage comp = TargetA.Thing.TryGetComp<CompPawnStorage>();
+            // Not sure why this is needed for animals and entities, but it shouldn't do any harm
+            CompAssignableToPawn_PawnStorage compAssignableToPawn = TargetA.Thing.TryGetComp<CompAssignableToPawn_PawnStorage>();
             if (ReleasingSpecific)
+            {
                 comp.ReleasePawn(TargetC.Pawn, ReleaseCell, actor.Map);
+                compAssignableToPawn.TryUnassignPawn(TargetC.Pawn);
```

Comment:
I don't think this is valid, because this will unassign a pawn from their designated statue every time they're released so they won't be able to go back in if it's schedulable and the user will have to reassign them every time. If it's causing some other issue we should fix that. I also thing it may NRE but as I don't think it should be here, that's not particularly important.

An alternate stand type without the assignment is a valid option or making it not use the bed-types and having our own including animal/entity but that's more of a refactor I suspect. This is a fine solution for the entities/animals as they can't be scheduled to go home anyway but needs to be restricted to only apply to entity/animals

Reply:
D'oh. Misunderstood how that worked. Makes much more sense now. Dropping this works just fine :)